### ISSUE TITLE
Page node data-id unnecessarily updates on keydown

### ISSUE THREAD

**Issue by jkhaui** [yuzhenmi/taleweaver#73]
I noticed on the demo there's some latency when typing; minor, but still noticeable. Of course there could be many reasons for this, but perhaps a contributing factor is that the id of the active page node seems to unnecessarily refresh itself every time a character is inserted

<img src="https://misc-ddocs.s3-ap-southeast-2.amazonaws.com/rendering.gif" />

**Comment by yuzhenmi** [comment]
Thanks for reporting the issue. I was making some fixes to the layout reflower, which breaks lines and pages, and I think this got addressed. I am not able to reproduce this anymore with the current demo. Would you be able to confirm?

**Comment by jkhaui** [comment]
@yuzhenmi great to hear! I would love to confirm, but an issue with the TypeScript loader is still blocking me from running this project locally. I'll try again later today.

If you were talking about the demo site at https://yuzhenmi.github.io/taleweaver/, I can still see the same behaviour being exhibited in Chrome's dev tools

**Comment by jkhaui** [comment]
By the way, I really like how thoroughly you've thought this project through. And writing it in TypeScript is great. Is there any reason you decided not to build TaleWeaver as an extension for an existing text editor? (E.g. ProseMirror, Slate, Draft.js, etc). 
And is there any way others can contribute to development? If there's anything I can do to help I'd be happy to, but at the moment I wouldn't have a clue where to begin

**Comment by yuzhenmi** [comment]
@jkhaui Yeah, I couldn't reproduce the issue on the demo site you linked, maybe there is some specific condition that I was missing.

I was using ProseMirror prior to starting this project. While ProseMirror is well-implemented, it, like all other established open source editors out there, uses `content-editable` to display document content and to capture user input. This means, the editor is naive of where things are rendered on the viewport. For most use-cases, this is OK. However, I wanted paginated documents, which requires breaking up pages at equal heights. This cannot be done efficiently with editors like ProseMirror because we need to know the position and height of each line in order to produce pages.

I don't have prior experience with building word processors, much of the project's design has been trial and error. I think the design so far has solved document serialization and layout reasonably well.

There are 2 features that I'm still trying to figure out, with no satisfactory solution so far:
- Pasting (https://github.com/yuzhenmi/taleweaver/issues/27)
- Text formatting (https://github.com/yuzhenmi/taleweaver/issues/6)

Then there are some more straight-forward tasks:
- Option to hide cursor completely when editor is out of focus: https://github.com/yuzhenmi/taleweaver/issues/48
- Key bindings for PC: https://github.com/yuzhenmi/taleweaver/issues/29
- Bindings for cursor-based text selection: https://github.com/yuzhenmi/taleweaver/issues/56

The project structure / development process has not been documented yet, I'll add something later. In the meantime, please feel free to shoot me an email if you have any questions!

**Comment by sunjingwu** [comment]
The condition that produce this issue is:
![p1](https://sunjingwu.oss-cn-shanghai.aliyuncs.com/others/page1.png)
![p2](https://sunjingwu.oss-cn-shanghai.aliyuncs.com/others/page2.png)
